contrib/docker-compose/standalone/haproxy/etc/haproxy.d/50_fe_http_handler.cfg

```diff
@@ -71,7 +71,12 @@ frontend fe_http_handler
     capture request header Content-Length len 10
 
     # Add the X-Forwarded-For header
+    .if !defined(HAP_BEHIND_PROXY) && !defined(HAP_BEHIND_CLOUDFLARE) && !defined(HAP_BEHIND_PROXY_EXCEPT_HKP)
```

Comment:
AFAICT this will set XFF even in the case where BEHIND_EXCEPT_HKP is set and the request comes in directly over 11371, is this intentional?

Reply:
The check rule should read `if any BEHIND is set, don't set XFF`. Port 11371 check earlier will enable/disable dynamically the CDN var, that on the other hand will make HAProxy pick the client IP directly or from XFF that's sent.

Reply:
If BEHIND_PROXY is set, it won't set XFF because it assumes XFF is already set upstream; that's as expected. If EXCEPT_HKP is set, it similarly won't set XFF here, but if the connection came in over 11371 then surely XFF *should* be set?

Would it not be better to test the txn.cdn/txn.cloudflare variables here instead of the BEHIND vars, since we already performed this logic in the stanza above?